### ISSUE TITLE
Getting KeyError while solving ode : dsolve(Eq(g(x).diff(x).diff(x) , (x-2)**2 +(x-3)**3), g(x)) 

### ISSUE THREAD

**Issue by sampadsaha5** [sympy/sympy#10867]
```
In  [91]: diffeq = Eq(g(x).diff(x).diff(x) , (x-2)**2 +(x-3)**3)

In  [92]: dsolve(diffeq, g(x))
Out [92]: 
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-92-1251d876c182> in <module>()
----> 1 k = dsolve(diffeq,g(x))

/home/ahappyidiot/anaconda2/lib/python2.7/site-packages/sympy/solvers/ode.pyc in dsolve(eq, func, hint, simplify, ics, xi, eta, x0, n, **kwargs)
    657             # The key 'hint' stores the hint needed to be solved for.
    658             hint = hints['hint']
--> 659             return _helper_simplify(eq, hint, hints, simplify)
    660 
    661 def _helper_simplify(eq, hint, match, simplify=True, **kwargs):

/home/ahappyidiot/anaconda2/lib/python2.7/site-packages/sympy/solvers/ode.pyc in _helper_simplify(eq, hint, match, simplify, **kwargs)
    679         # attempt to solve for func, and apply any other hint specific
    680         # simplifications
--> 681         sols = solvefunc(eq, func, order, match)
    682         free = eq.free_symbols
    683         cons = lambda s: s.free_symbols.difference(free)

/home/ahappyidiot/anaconda2/lib/python2.7/site-packages/sympy/solvers/ode.pyc in ode_nth_linear_constant_coeff_undetermined_coefficients(eq, func, order, match)
   4726         returns='both')
   4727     match.update(gensol)
-> 4728     return _solve_undetermined_coefficients(eq, func, order, match)
   4729 
   4730 

/home/ahappyidiot/anaconda2/lib/python2.7/site-packages/sympy/solvers/ode.pyc in _solve_undetermined_coefficients(eq, func, order, match)
   4819     for i in Add.make_args(eqs):
   4820         s = separatevars(i, dict=True, symbols=[x])
-> 4821         coeffsdict[s[x]] += s['coeff']
   4822 
   4823     coeffvals = solve(list(coeffsdict.values()), coefflist)

KeyError: (x - 3)**3
```

**Comment by smichr** [comment]
`expand` is probably overkill. How about the following:

```python
>>> from sympy.core.function import _mexpand
>>> Eq(x,(x+1)**2)
Eq(x, (x + 1)**2)
>>> _.func(*[expand_mul(a) for a in _.args])
Eq(x, (x + 1)**2)  # yep, it fails because a multinomial is not considered a Mul
>>> _mexpand(_)
Eq(x, x**2 + 2*x + 1)
```

See the docstring on `_mexpand` and consider whether you should use the `recursive` flag.

**Comment by bjodah** [comment]
Fixed in #10873
